src/raytracing/compute_phong.c

```
@@ -0,0 +1,75 @@
+    // 光の方向を計算（光源位置 - 交点）
+    light_dir = normalize(subtract(scene->light.position, hit_point));
+    
+    // diffuse (Lambert)
+    diffuse = fmax(dot_product(normal, light_dir), 0.0);
+
+    // specular (Phong)
+    reflected_dir = reflect(scale(light_dir - 1), normal); // R = 2N(L·N) - L
+    specular = pow(fmax(dot_product(reflected_dir, normalize(view_dir)), 0.0),
+            shiness); // shininess = 50
+
+    color.r = ambient * object_color.r
```

Cppcheck (2.10-2)

[error] Uninitialized variable

(uninitvar)